src/lib.rs

```diff
@@ -0,0 +1,450 @@
+extern crate sxd_document;
+
+use std::str;
+use std::string::String;
+use std::collections::HashMap;
+
+use sxd_document::{
+    parser,
+    Package,
+};
+use sxd_document::dom::{
+    Document,
+    Root,
+    ChildOfRoot,
+    Element,
+    ChildOfElement,
+};
+
+#[derive(Debug)]
+pub struct Program {
+    pub groups: Vec<StatementBody>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct StatementBody {
+    pub blocks: Vec<Block>
+}
+
+#[derive(PartialEq, Debug)]
+pub struct Block {
+    pub block_type: String,
+    pub id: String,
+    pub fields: Option<HashMap<String, FieldValue>>,
+    pub statements: Option<HashMap<String, StatementBody>>,
+}
+
+#[derive(PartialEq, Debug)]
+pub enum FieldValue {
+    SimpleField(String),
+    ExpressionField(Block),
+}
+
+impl Program {
```

Comment:
Is it more conventional to group `struct`s together, then `impl`s, or to group `struct` and `impl` blocks for each entity?